src/test/ui/higher-rank-trait-bounds/issue-59311.stderr

```diff
@@ -1,8 +1,10 @@
-error: higher-ranked subtype error
+error: higher-ranked lifetime error
   --> $DIR/issue-59311.rs:17:9
    |
 LL |     v.t(|| {});
    |         ^^^^^
+   |
+   = note: could not prove for<'a> &'a V: 'static
```

Comment:
Ideally we would point at the `where` clause as well. (not for this PR)

Reply:
This should not actually be an error - see https://github.com/rust-lang/rust/issues/89409. However, it occurs both with and without `#![feature(nll)]`, so I don't think we need to address this before turning on NLL mode.